### ISSUE TITLE
Add tile option

### ISSUE THREAD

**Issue by marcobarbierato** [muslll/neosr#63]
Real-ESRGAN and HAT-GAN offer the option `tile` in `yml` files when doing inference. See for example:

https://github.com/XPixelGroup/HAT/blob/c7e0b2b9c9a8d37a3cf3e7dfa46698d0507d8f1f/options/test/HAT_GAN_Real_SRx4.yml#L1-L11

```
name: HAT_GAN_Real_SRx4
model_type: HATModel
scale: 4
num_gpu: 1  # set num_gpu: 0 for cpu mode
manual_seed: 0

tile: # use the tile mode for limited GPU memory when testing.
  tile_size: 512 # the higher, the more utilized GPU memory and the less performance change against the full image. must be an integer multiple of the window size.
  tile_pad: 32 # overlapping between adjacency patches.must be an integer multiple of the window size.

datasets:
  test_1:  # the 1st test dataset
    name: custom
    type: SingleImageDataset
    dataroot_lq: input_dir
    io_backend:
      type: disk
```

EDIT: I've seen that there's already something commented in the code:
https://github.com/muslll/neosr/blob/ff23dc06ece18589c93755f8fac36416ab70ca71/neosr/models/default.py#L472-L474

It doesn't make use of custom tile size or pad, but out of the box it does work for RealESRGAN.

**Comment by muslll** [comment]
Hi. Yes, this is on my todo list. As you pointed out, I still need to verify if the function works on edge cases, hence why I didn't make this option available yet. I'll post it here when I do so.

**Comment by muslll** [comment]
Fixed on [15e1a2d](https://github.com/muslll/neosr/commit/15e1a2dd15f646c8efbf3a1f7ecd943088b65203).
